packages/@ember/-internals/routing/lib/system/router.ts

```diff
@@ -381,23 +379,6 @@ class EmberRouter extends EmberObject.extend(Evented) implements Evented {
       }
 
       didTransition(infos: PrivateRouteInfo[]) {
-        if (ROUTER_EVENTS) {
-          if (router.didTransition !== defaultDidTransition) {
-            deprecate(
-              'You attempted to override the "didTransition" method which is deprecated. Please inject the router service and listen to the "routeDidChange" event.',
-              false,
-              {
-                id: 'deprecate-router-events',
-                until: '4.0.0',
-                url: 'https://deprecations.emberjs.com/v3.x#toc_deprecate-router-events',
-                for: 'ember-source',
-                since: {
-                  enabled: '3.11.0',
-                },
-              }
-            );
-          }
-        }
         router.didTransition(infos);
```

Comment:
Is this the correct way to deal with this? Or do these methods go away entirely?

Reply:
@knownasilya My initial feeling is we need to make `defaultDidTransition` && `defaultWillTransition` the defacto method that is called when didTransition/willTransition is called on the PrivateRouter.  I'm wondering if we can take everything in, say `defaultDidTransition` and just throw it in here.  What do you think?